extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -87,12 +81,27 @@ public Object get(ByteBuffer buf, int position)
   //Note that this is not threadsafe and I don't think it needs to be
   private Union getUnion(ByteBuffer buf, int position)
   {
-    Union union = unions.get(position);
-    if (union == null) {
-      Memory mem = new MemoryRegion(nm, position, maxIntermediateSize);
-      union = (Union) SetOperation.wrap(mem);
-      unions.put(position, union);
+    Int2ObjectMap<Union> unionMap = unions.get(buf);
+    Union union = unionMap != null ? unionMap.get(position) : null;
+    if (union != null) {
+      return union;
+    }
+    return createNewUnion(buf, position, true);
+  }
+
+  private Union createNewUnion(ByteBuffer buf, int position, boolean isWrapped)
+  {
+    NativeMemory nm = getNativeMemory(buf);
+    Memory mem = new MemoryRegion(nm, position, maxIntermediateSize);
+    Union union = isWrapped
+                  ? (Union) SetOperation.wrap(mem)
+                  : (Union) SetOperation.builder().initMemory(mem).build(size, Family.UNION);
+    Int2ObjectMap<Union> unionMap = unions.get(buf);
```

Comment:
@leventov BufferAggregator interface does not tell about the order of execution mentioned here https://github.com/druid-io/druid/pull/4071#discussion_r107329220, now BufferAggregators implementations have two options:
1. Favor correctness : i.e to support any ByteBuffer in aggregate-related methods even without `init()` being called for some ByteBuffer.
2. Favor performance : i.e assuming https://github.com/druid-io/druid/pull/4071#discussion_r107329220 and fail aggregate-related methods if it sees a new ByteBuffer without `init()`.

I choose correctness over performance and Its unfair to block someones PR who disagrees with your choice.

FYI: @himanshug already +1'ed this PR.